clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThreadTest.java

```diff
@@ -258,155 +252,66 @@ void testPollResultTimer() {
-
-        Node node = metadata.fetch().nodes().get(0);
-        coordinatorRequestManager.markCoordinatorUnknown("test", time.milliseconds());
-        client.prepareResponse(FindCoordinatorResponse.prepareResponse(Errors.NONE, "group-id", node));
-        prepareOffsetCommitRequest(new HashMap<>(), Errors.NONE, false);
-        CompletableApplicationEvent<Void> event1 = spy(new AsyncCommitEvent(Collections.emptyMap()));
-        ApplicationEvent event2 = new AsyncCommitEvent(Collections.emptyMap());
-        CompletableFuture<Void> future = new CompletableFuture<>();
-        when(event1.future()).thenReturn(future);
-        applicationEventsQueue.add(event1);
-        applicationEventsQueue.add(event2);
-        assertFalse(future.isDone());
-        assertFalse(applicationEventsQueue.isEmpty());
-        consumerNetworkThread.cleanup();
-        assertTrue(future.isCompletedExceptionally());
-        assertTrue(applicationEventsQueue.isEmpty());
+        // After runOnce has been called, it takes the default heartbeat interval from the heartbeat request manager
+        assertEquals(DEFAULT_HEARTBEAT_INTERVAL_MS, consumerNetworkThread.maximumTimeToWait());
     }
 
     @Test
-    void testCleanupInvokesReaper() {
+    public void testCleanupInvokesReaper() {
+        LinkedList<NetworkClientDelegate.UnsentRequest> queue = new LinkedList<>();
+        when(networkClientDelegate.unsentRequests()).thenReturn(queue);
         consumerNetworkThread.cleanup();
         verify(applicationEventReaper).reap(applicationEventsQueue);
     }
 
     @Test
-    void testRunOnceInvokesReaper() {
+    public void testRunOnceInvokesReaper() {
         consumerNetworkThread.runOnce();
         verify(applicationEventReaper).reap(any(Long.class));
     }
 
     @Test
-    void testSendUnsentRequest() {
-        String groupId = "group-id";
-        NetworkClientDelegate.UnsentRequest request = new NetworkClientDelegate.UnsentRequest(
-            new FindCoordinatorRequest.Builder(
-                new FindCoordinatorRequestData()
-                    .setKeyType(FindCoordinatorRequest.CoordinatorType.TRANSACTION.id())
-                    .setKey(groupId)),
-            Optional.empty());
-
-        networkClient.add(request);
-        assertTrue(networkClient.hasAnyPendingRequests());
-        assertFalse(networkClient.unsentRequests().isEmpty());
-        assertFalse(client.hasInFlightRequests());
+    public void testSendUnsentRequests() {
+        when(networkClientDelegate.unsentRequests()).thenReturn(new LinkedList<>());
+        when(networkClientDelegate.hasAnyPendingRequests()).thenReturn(true).thenReturn(true).thenReturn(false);
+
         consumerNetworkThread.cleanup();
 
-        assertTrue(networkClient.unsentRequests().isEmpty());
+        verify(networkClientDelegate, times(2)).poll(anyLong(), anyLong());
+        assertTrue(networkClientDelegate.unsentRequests().isEmpty());
         assertFalse(client.hasInFlightRequests());
```

Comment:
```suggestion
        verify(networkClientDelegate, times(2)).poll(anyLong(), anyLong());
```

We don't need to assert on the mocked clients. As long as we can verify that the consumer network thread is calling the client poll, the rest is the responsibility of the networkClientDelegate

Reply:
Makes sense, fixed